### PR TITLE
Remove Flood options from rtorrentvpn

### PR DESCRIPTION
**Pull request by BanditTech**
# Pull request

**Purpose**
remove flood options from rtorrentvpn

**Approach**
it fixes it with code!

I found something interesting while digging around trying to get flood working inside the rtorrentvpn container...
The documentation on dockstarter is saying that the package includes flood, and includes options and ports which seem to be for it.
But if you look at the package which it is pulling the image from:
https://hub.docker.com/r/binhex/arch-rtorrentvpn/
Then you will notice there is no longer any mention of Flood coming packaged within this container.

```
docker run -d \
    --cap-add=NET_ADMIN \
    -p 9080:9080 \
    -p 9443:9443 \
    -p 8118:8118 \
    --name=<container name> \
    -v <path for data files>:/data \
    -v <path for config files>:/config \
    -v /etc/localtime:/etc/localtime:ro \
    -e VPN_ENABLED=<yes|no> \
    -e VPN_USER=<vpn username> \
    -e VPN_PASS=<vpn password> \
    -e VPN_PROV=<pia|airvpn|custom> \
    -e VPN_CLIENT=<openvpn|wireguard> \
    -e VPN_OPTIONS=<additional openvpn cli options> \
    -e STRICT_PORT_FORWARD=<yes|no> \
    -e ENABLE_PRIVOXY=<yes|no> \
    -e ENABLE_AUTODL_IRSSI=<yes|no> \
    -e ENABLE_RPC2=<yes|no> \
    -e ENABLE_RPC2_AUTH=<yes|no> \
    -e ENABLE_WEBUI_AUTH=<yes|no> \
    -e RPC2_USER=<rpc2 username> \
    -e RPC2_PASS=<rpc2 password> \
    -e WEBUI_USER=<webui username> \
    -e WEBUI_PASS=<webui password> \
    -e LAN_NETWORK=<lan ipv4 network>/<cidr notation> \
    -e NAME_SERVERS=<name server ip(s)> \
    -e ADDITIONAL_PORTS=<port number(s)> \
    -e DEBUG=<true|false> \
    -e PHP_TZ=<php timezone> \
    -e UMASK=<umask for created files> \
    -e PUID=<uid for user> \
    -e PGID=<gid for user> \
    binhex/arch-rtorrentvpn
```
so there are several changed options, most notably the removal of several port forwards
```
      - ${RTORRENTVPN_PORT_3000}:3000
      - ${RTORRENTVPN_PORT_51413}:51413
      - ${RTORRENTVPN_PORT_51413}:51413/udp
      - ${RTORRENTVPN_PORT_6881}:6881
      - ${RTORRENTVPN_PORT_6881}:6881/udp
```
```
      - ENABLE_FLOOD=${RTORRENTVPN_ENABLE_FLOOD}
```
have been removed

### CODE REVIEW

**Comment by nemchik** [comment]
It makes sense to remove the flood options, but I wasn't sure about the rest of the changes. The strict port option didn't seem necessary, but I'm fine including it. We should include it in the other VPN downloader images (the ones with VPN in the name) as well.

I think the ports were included as they seemed standard and necessary for tTorrent, but I'm not against removing them if they aren't needed.

Keep in mind, we would need to remove these things in multiple places. Mainly the labels file, ports file, and the main file (named for the app) environment section.

**Comment by BanditTech** [comment]
I just included it because I thought it was a new setting, but after looking further I think you are right. It is not really required for most people.

Mostly it would just be removing the stale settings which are no longer present in the image. 

But after testing the different torrent downloaders and their GUI, this just feels a bit old and clunky compared to qBitTorrent. So I would not recommend anyone to use this in a new install compared to qBit... With so many choices of downloaders, for new people it might be better to just remove some of the poorer choices IMO.

**Comment by BanditTech** [comment]
the files edited so far are inside the .app folder, the three you mentioned I thought I did. Did I miss something?

**Comment by BanditTech** [comment]
the old ports were for the flood client, which is no longer packaged in this image

**Comment by nemchik** [comment]
I don't think this image is broken, but there do seem to be some deprecated options (flood) still lingering in our template.

As for including it in DS, the goal has never been to limit what options users have. Letting users choose what to run is a bit of a core goal of the project. I personally agree qbit is better, but if someone wants to run something else, it's up to them.

**Comment by BanditTech** [comment]
That is true, its good to give choices. 
It is just unfortunate that the best torrent option shows up after every single other worse choice on the list, and there are several.

**Comment by nemchik** [comment]
> That is true, its good to give choices.
> It is just unfortunate that the best torrent option shows up after every single other worse choice on the list, and there are several.

Yeah, it's just alphabetical. I'll submit a proper review with inline comments for anything else this needs.

File: compose/.apps/rtorrentvpn/rtorrentvpn.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ services:
     container_name: rtorrentvpn
     environment:
       - ENABLE_AUTODL_IRSSI=${RTORRENTVPN_ENABLE_AUTODL_IRSSI}
-      - ENABLE_FLOOD=${RTORRENTVPN_ENABLE_FLOOD}
+      - STRICT_PORT_FORWARD=${RTORRENTVPN_STRICT_PORT_FORWARD}
```
Comment (nemchik) [inline compose/.apps/rtorrentvpn/rtorrentvpn.yml:9]: Let's skip adding the strict variable. Removing the flood variable is fine.

File: compose/.apps/rtorrentvpn/rtorrentvpn.labels.yml [hunk 2]
```diff
@@ -5,11 +5,11 @@ services:
...
       com.dockstarter.appvars.rtorrentvpn_enable_autodl_irssi: "no"
-      com.dockstarter.appvars.rtorrentvpn_enable_flood: both
       com.dockstarter.appvars.rtorrentvpn_enable_privoxy: "no"
       com.dockstarter.appvars.rtorrentvpn_enable_rpc2: "yes"
       com.dockstarter.appvars.rtorrentvpn_enable_rpc2_auth: "yes"
       com.dockstarter.appvars.rtorrentvpn_enable_webui_auth: "yes"
+      com.dockstarter.appvars.rtorrentvpn_strict_port_forward: "yes"
       com.dockstarter.appvars.rtorrentvpn_enabled: "false"
       com.dockstarter.appvars.rtorrentvpn_network_mode: ""
       com.dockstarter.appvars.rtorrentvpn_port_3000: "3000"
```
Comment (nemchik) [inline compose/.apps/rtorrentvpn/rtorrentvpn.labels.yml:15]: Let's remove the 3 unused ports from this file as well.

File: compose/.apps/rtorrentvpn/rtorrentvpn.labels.yml [hunk 3]
```diff
@@ -5,11 +5,11 @@ services:
       com.dockstarter.appinfo.description: rTorrent, Flood or ruTorrent WebUI, OpenVPN and Privoxy
       com.dockstarter.appinfo.nicename: rTorrentVPN
       com.dockstarter.appvars.rtorrentvpn_enable_autodl_irssi: "no"
-      com.dockstarter.appvars.rtorrentvpn_enable_flood: both
       com.dockstarter.appvars.rtorrentvpn_enable_privoxy: "no"
       com.dockstarter.appvars.rtorrentvpn_enable_rpc2: "yes"
       com.dockstarter.appvars.rtorrentvpn_enable_rpc2_auth: "yes"
       com.dockstarter.appvars.rtorrentvpn_enable_webui_auth: "yes"
+      com.dockstarter.appvars.rtorrentvpn_strict_port_forward: "yes"
```
Comment (nemchik) [inline compose/.apps/rtorrentvpn/rtorrentvpn.labels.yml:12]: Let's not add the strict variable in this PR. We could consider adding it to all the VPN download client containers in a separate pr.

**Review by nemchik — CHANGES_REQUESTED** [review]

File: compose/.apps/rtorrentvpn/rtorrentvpn.yml (hunk 1 above)
Comment (BanditTech) [reply compose/.apps/rtorrentvpn/rtorrentvpn.yml]: thats a good idea

**Comment by BanditTech** [comment]
OK, sorry for the delay, I have been a little preoccupied. Thank you for the notes, I followed them and feel its ready for merge now.

**Comment by nemchik** [comment]
I need to have another look at this.

**Comment by nemchik** [comment]
Awkward timing, binhex deprecated the image a few days ago https://github.com/binhex/arch-rtorrentvpn/commit/98351f05c07600691c30a99473ccf3475c5b5121

I believe `51413` and `6881` are meant for incoming connections, and not documented by binhex.

Aside from that, I believe your changes are correct, so I'm going to merge them, but I think we will likely mark this app as deprecated in the future.

**Review by nemchik — APPROVED** [review]

**Pull request merged**